Assets/Scripts/Combat/Abilities/MagicShield.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class MagicShield : Ability
+{
+    private enum Button { Up, Down, Left, Right };
+    private enum Phase { SequencePhase, InputSequence, Activate, Inactive };
+    private Timer SequenceTimer;
+    private Queue<Button> Sequence = new Queue<Button>();
+    private Queue<Button> SequenceOrder = new Queue<Button>();
+    private Queue<GameObject> Directions = new Queue<GameObject>();
+
+    [SerializeField]
+    private GameObject Shield;
+    [SerializeField]
+    private GameObject[] Arrows; // 0: Up, 1: Down, 2: Left, 3: Right 
+
+    private readonly float SequenceDuration = 2f;
+    private readonly float InputDuration = 5f;
+    private readonly float PositionOffset = 4f;
+    private Vector3 Position = Vector3.zero;
+    private Phase CurrentPhase = Phase.Inactive;
+
+    private int CorrectInputs = 0;
+    private readonly int MagicShieldFactor = 20;
+    private int MagicShieldHealth;
+
+    public new void Start()
+    {
+        base.Start();
+        SequenceTimer = GetComponent<Timer>();
+        Shield.SetActive(false);
+
+        foreach(GameObject arrow in Arrows)
+            arrow.SetActive(false);
+
+        TargetSchema = new TargetSchema(
+            0,
+            CombatantType.Ally,
+            SelectorType.All);
+    }
+
+    new public void StartAbility()
+    {
+        CorrectInputs = 0;
+        base.StartAbility();
+    }
+
+    private void Update()
+    {
+        if(CurrentPhase == Phase.SequencePhase)
+            SequencePhaseUpdate();
+
+        else if(CurrentPhase == Phase.InputSequence)
+            InputSequenceUpdate();
+    }
+
+    protected override void ContinueAbilityAfterTargeting()
+    {
+        StartMagicShieldSequence();
+    }
+
+    protected override void EndAbility()
+    {
+        Debug.Log($"Magic Shield Total Health: {MagicShieldHealth}");
+
+        for (int i = 0; i < TargetedCombatants.Length; i++)
+            TargetedCombatants[i].GetComponent<Combatant>().ApplyShield(MagicShieldHealth);
+
+        CombatSystem.EndTurn(this.GetComponentInParent<Combatant>().gameObject);
+
+    }
+
+    private void StartMagicShieldSequence()
+    {
+        Debug.Log("Starting Magic Shield Ability");
+
+        Shield.transform.position = GameObject.FindGameObjectWithTag("Player").transform.position;
+        Shield.transform.position -= new Vector3(0, 5, 0);
+        Shield.transform.Translate(new Vector3(0, 10, 0));
```

Comment:
I move the shield prefab to wherever the player is. Then, I translate it upwards slightly so it covers the players. What I could do instead is put it in the Combatants class, since that is where the shield health is being dealt with.